doc/examples/applications/plot_3d_segmentation_embryo.py

```diff
@@ -0,0 +1,206 @@
+"""
+==================================================
+Segment 3D image sample of developing mouse embryo
+==================================================
+
+In this example, we look at a microscopy image of a developing mouse embryo.
+We use sample data from [1]_, more precisely from embryo B at time point 184.
+
+.. [1] McDole K, Guignard L, Amat F, Berger A, Malandain G, Royer LA,
+       Turaga SC, Branson K, Keller PJ (2018) "In Toto Imaging and
+       Reconstruction of Post-Implantation Mouse Development at the
+       Single-Cell Level" Cell, 175(3):859-876.e33.
+       ISSN: 0092-8674
+       :DOI:`10.1016/j.cell.2018.09.031`
+
+"""
+
+import io
+import requests
+
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import ndimage as ndi
+
+import skimage as ski
+
+
+#####################################################################
+# We downloaded the original data in KLB format and sliced a particular
+# subvolume, which we saved into a compressed Numpy format:
+#
+# .. code-block:: python
+#
+#     import numpy as np
+#     import pyklb as klb
+#
+#     data = klb.readfull('Mmu_E1_CAGTAG1.TM000184_timeFused_blending/SPM00_TM000184_CM00_CM01_CHN00.fusedStack.corrected.shifted.klb')
+#     sample = data[400:450, 1000:1750, 400:900]
+#     np.savez_compressed('sample_3D_frame_184.npz', sample)
+
+#####################################################################
+# View 3D image data
+# ==================
+
+sample_url = 'https://gitlab.com/scikit-image/data/-/raw/30a6bf082e5a91a2ee97e003465537224ffad216/Embryo2/sample_3D_frame_184.npz'
+response = requests.get(sample_url)
+im3d_dict = np.load(io.BytesIO(response.content))
+im3d = im3d_dict['arr_0']
+
+print(f'The shape of the image is: {im3d.shape}')
+
+#####################################################################
+# The sample dataset is a 3D image with 50 `xy` sections stacked along `z`. Let us
+# visualize it by picking one such section in five.
+
+data_montage = ski.util.montage(im3d[::5], grid_shape=(2, 5), padding_width=5)
+fig, ax = plt.subplots(figsize=(10, 5))
+ax.imshow(data_montage)
+ax.set_axis_off()
+
+#####################################################################
+# Apply thresholding techniques
+# =============================
+
+# global thresholding vs local thresholding
+global_thresh = ski.filters.threshold_otsu(im3d)
+binary_global = im3d > global_thresh
+
+block_size = 31
+local_thresh = ski.filters.threshold_local(im3d, block_size)
+binary_local = im3d > local_thresh
+
+#####################################################################
+# Let us view the mid-stack `xy` section.
+
+fig, axes = plt.subplots(ncols=3, figsize=(12, 4))
+ax = axes.ravel()
+
+ax[0].imshow(im3d[25, :, :])
+ax[0].set_title('Original')
+
+ax[1].imshow(binary_global[25, :, :])
+ax[1].set_title('Global thresholding (Otsu)')
+
+ax[2].imshow(binary_local[25, :, :])
+ax[2].set_title('Local thresholding')
```

Comment:
Due to the default [interpolation](https://matplotlib.org/stable/gallery/images_contours_and_fields/interpolation_methods.html) (`'antialiased'`), the rendered figures give the illusion that the data isn't binary, e.g. intermediate green hues are used. Not sure which is the best interpolation option here but both "none" and "nearest" seem to preserve the binary look

```suggestion
ax[1].imshow(binary_global[25, :, :], interpolation="nearest")
ax[1].set_title('Global thresholding (Otsu)')

ax[2].imshow(binary_local[25, :, :], interpolation="nearest")
ax[2].set_title('Local thresholding')
```

Same applies to other similar plots...